src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -237,52 +237,76 @@ class DefaultColorControlSceneHandler : public scenes::DefaultSceneHandlerImpl
             case Attributes::CurrentX::Id:
                 if (SupportsColorMode(endpoint, ColorControl::EnhancedColorMode::kCurrentXAndCurrentY))
                 {
-                    if (decodePair.attributeValue)
+                    if (decodePair.valueUnsigned16.HasValue())
```

Comment:
Spec says "SHALL be considered invalid" in this case.  Doesn't that mean there should be some errors returned somewhere when such an invalid struct is presented?

Followup to figure that out.

Reply:
Just to make sure, you mean in the case where we do not have a value in valueUnsigned16? 

If so I added a `VerifyOrReturnError` check for `HasValue`.